File: pascalsTriangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var triangle = new List<IList<int>>();

        for (int row = 0; row < numRows; row++) {
            var currentRow = new List<int>();
            currentRow.Add(1); // First element is always 1

            for (int col = 1; col < row; col++) {
                int sum = triangle[row - 1][col - 1] + triangle[row - 1][col];
                currentRow.Add(sum);
            }

            if (row > 0)
                currentRow.Add(1); // Last element is also 1

            triangle.Add(currentRow);
        }

        return triangle;
    }
}

File: pascalsTriangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []

        triangle = [[1]]
        for i in range(1, numRows):
            prev_row = [0] + triangle[-1] + [0]
            current_row = [prev_row[j] + prev_row[j+1] for j in range(i + 1)]
            triangle.append(current_row)
        return triangle
